src/context/mod.rs

```diff
@@ -0,0 +1,180 @@
+use ckb_vm_definitions::instructions::Instruction;
+
+use super::Error;
+use crate::{machine::SupportMachine, syscalls::Syscalls, Debugger};
+
+pub trait ExecutionContext<Mac: SupportMachine> {
+    fn initialize(&mut self, machine: &mut Mac) -> Result<(), Error> {
+        // We don't want to change the param name to start with an
+        // underscore(_). It doesn't look good in docs. Also when someone
+        // implements this method with IDE completion, they will need to remove
+        // the staring underscore.
+        #[allow(clippy::drop_ref)]
+        drop(machine);
+        Ok(())
+    }
+    /// Return true if the syscall has been processed. If a module returns
```

Comment:
suggest to use `allow unused_variables` in trait level:
```suggestion
#[allow(unused_variables)]
pub trait ExecutionContext<Mac: SupportMachine> {
    fn initialize(&mut self, machine: &mut Mac) -> Result<(), Error> {
        Ok(())
    }
```

Reply:
I don't think this is better because it's allowing more stuff than necessary and the intent is less clear.

This [comment](https://internals.rust-lang.org/t/unused-parameters-in-default-trait-impls/10984/7) suggests using #[allow(unused_variables)] on individual parameters, but IMO it makes the signature a lot messier.

Reply:
rust team member suggest to use unused_variables: https://github.com/rust-lang/rust/issues/26487#issuecomment-169828205

Reply:
It's suggesting `#[allow]`, and we are using `#[allow]` on the most specific item.